include/indexstore/indexstore.h

```diff
@@ -53,9 +53,13 @@
 
 #ifndef INDEXSTORE_PUBLIC
 # if defined (_MSC_VER)
-#  define INDEXSTORE_PUBLIC __declspec(dllimport)
+#  if defined(_CINDEX_LIB_)
+#    define INDEXSTORE_PUBLIC __declspec(dllexport)
+#  else
+#    define INDEXSTORE_PUBLIC __declspec(dllimport)
+#  endif
 # else
-#  define INDEXSTORE_PUBLIC
+#  define INDEXSTORE_PUBLIC __attribute__((__visibility__("public")))
```

Comment:
Why is this change necessary ?

Reply:
It's not strictly necessary, but does mean that we can actually remove the exports list (which is problematic for Windows, since we need to ensure that all symbols listed are defined, which is not the case when built without blocks).

Reply:
The export list is useful to ensure that only the listed symbols are exported and nothing else. Any other symbols would be prevented from being exported, making them private details of the library. I don't think it's a good idea to remove the export list.

Reply:
Right, with the suggested change, only the symbols marked with `INDEXSTORE_PUBLIC` would be exported and nothing else.  Okay,  I suppose that we can disable the exports list on Windows then.

Reply:
I thought for some reason that Windows doesn't use the export list (that it's unix specific) and we get things exported by marking them `__declspec(dllexport)` ?
If windows does use the export list I'd be fine with changing windows to not use it.

Reply:
Yeah, Windows does get it through the `__declspec(dllexport)` (on ELF and MachO targets, `__attribute__((__visibility__("default")))` with `-fvisibility-default=hidden` and `-fvisibility-inlines-hidden` will accomplish similar behaviour).  At that point, the source contains the actual source of truth.

Reply:
You would have to apply the new compiler options to _everything_, including llvm and clang, not just swift. And if then you'd have to convince everyone to mark explicitly the symbols that they want exported.
I don't think it is worth it, I generally like the greater control that the export list give us, there's a specific list of things to export and nothing else.